src/main/java/de/sakpaas/backend/service/SearchService.java

```diff
@@ -97,6 +101,30 @@ protected SearchRequest dbBrandSearch(SearchRequest request) {
    * @return the updated Request Object
    */
   protected SearchRequest getByCoordinates(SearchRequest request) {
+    // Get Locations
+    CoordinateDetails coordinateDetails = request.getCoordinates();
+    List<Location> locations = locationService
+        .findByCoordinates(coordinateDetails.getLatitude(), coordinateDetails.getLongitude());
+
+    // Filter by brand
+    if (knownBrands.isEmpty()) {
+      request.setLocations(locations);
+    } else {
+      List<Location> locationList = new ArrayList<>();
+      for (String brand : knownBrands) {
+        locationList.addAll(
+            locations.stream()
+                .filter(location -> location.getName().contains(brand))
+                .collect(Collectors.toList()));
+
+        locationList.addAll(
+            locations.stream()
+                .filter(location -> location.getDetails().getBrand().equals(brand))
+                .collect(Collectors.toList()));
+      }
+      request.setLocations(locationList);
```

Comment:
Wenn ich mich nicht täusche, kann es sein dass Einträge doppelt in der Liste sind. Wenn z.B. eine Location den Namen "Lidl Mannheim" hat und als Brand auch "Lidl".